Cleanups

krnowak opened the pull request:
These are some cleanups and refactorings I did when working on avahi stuff. Probably best reviewed commit-by-commit.

@iaguis: You wanted to see this.

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -160,127 +153,136 @@ static const gchar *dmi_attributes[] =
+  return g_object_ref (booted_deployment);
+}
+
+static gchar *
```
Does this really need to duplicate the string? It looks to me that the only client that needs this duplicated (the hash table code below) could do it there instead.

krnowak replied:
The `eos_updater_get_booted_checksum` also uses this function.

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -160,127 +153,136 @@ static const gchar *dmi_attributes[] =
-  osname = ostree_deployment_get_osname (merge_deployment);
-  origin = ostree_deployment_get_origin (merge_deployment);
-  booted = ostree_deployment_get_csum (merge_deployment);
+  return g_object_ref (booted_deployment);
```
Does this really need to add a ref? It does not look like the clients really have that requirement

krnowak replied:
`ostree_sysroot_get_booted_deployment` does not transfer the ownership, so if we don't ref it, the booted deployment might be deleted when we unref the sysroot. It used to work without adding a ref because sysroot was actually leaking its deployments earlier (https://github.com/ostreedev/ostree/pull/287).

cosimoc replied:
Oh, good point.

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -291,17 +293,25 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
-  message ("Product group: %s", p_id);
+  *sys_vendor = g_strdup (g_hash_table_lookup (hw_descriptors, "sys_vendor"));
+  *product_name = g_strdup (g_hash_table_lookup (hw_descriptors, "product_name"));
+  return hw_descriptors;
```
I wonder if you really need to also return these values here separately, instead of looking them up from the table where you need them

krnowak replied:
I wanted to contain the use of the magic strings like "sys_vendor" and "product_name" just in one place, but I guess I can hide those names behind some `#define`.

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -313,73 +323,180 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
+                      gchar **p_ref,
+                      GError **error)
+{
+  if (g_key_file_get_boolean (bkf, group_name, ON_HOLD_KEY, NULL))
```
Pass the error here

krnowak replied:
Oops, thanks for spotting.

krnowak replied:
Hm, looks like we have to handle it a bit more specially - the OnHold key might be missing and that's fine, we don't want to propagate the error in this case, right?

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -313,73 +323,180 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
-      p_ref = g_key_file_get_string (bkf, DEFAULT_GROUP, OSTREE_REF_KEY, error);
-      if (!p_ref)
-        goto out;
+      if (!process_single_group (bkf, group_name, on_hold, p_ref, error))
```
Should be `DEFAULT_GROUP` instead of `group_name`

krnowak replied:
Oops. Will fix.

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -313,73 +323,180 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
+{
+  static gboolean metric_sent = FALSE;
+
+  if (G_LIKELY (metric_sent))
```
We usually don't use `G_LIKELY` and `G_UNLIKELY` because they make the code more complicated and their usefulness is questionable... Please remove

krnowak replied:
Sure.

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -313,73 +323,180 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
+    return FALSE;
+
+  if (!process_branch_file (bkf, product_group, &on_hold, &ref, error))
+    return FALSE;
```
Add newline

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -313,73 +323,180 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
+      message ("Using product branch %s", ref);
+      *upgrade_refspec = g_strdup_printf ("%s:%s", booted_remote, ref);
+      *original_refspec = g_strdup_printf ("%s:%s", booted_remote, booted_ref);
+    }
```
Add newline

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -313,73 +323,180 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
+  booted_deployment = get_booted_deployment (error);
+  if (booted_deployment == NULL)
+    return FALSE;
+  return get_upgrade_info (repo,
```
Add newline

cosimoc commented on src/eos-updater-util.h:
```diff
@@ -37,21 +37,29 @@
 
 #define EOS_UPDATER_ERROR (eos_updater_error_quark())
-GQuark eos_updater_error_quark (void);
+GQuark
```
It's normal for headers to have the return type on the same line in GObject style code.

krnowak replied:
I had an impression that the types were always in a separate line in gobject, but I see I was wrong. Will revert it.

cosimoc commented on src/eos-updater.c:
```diff
@@ -85,6 +85,7 @@ on_bus_acquired (GDBusConnection *connection,
 
   /* We are deliberately not emitting a signal here. This
    * isn't a state change, it's our initial state.
+   * krnowak: ORLY? It calls eos_updater_emit_state_changed…
```
lol

cosimoc commented on src/eos-updater-apply.c:
```diff
@@ -78,7 +78,6 @@ apply (GTask *task,
   GMainContext *task_context = g_main_context_new ();
   const gchar *update_id = eos_updater_get_update_id (updater);
   const gchar *update_refspec = eos_updater_get_update_refspec (updater);
-  const gchar *orig_refspec = eos_updater_get_original_refspec (updater);
```
I don't really understand this commit - the code is not exporting the property just to propagate it, but it "stores" it locally on the `GDBusProxy` object not differently from how you store it in the updater data. The property is separately exported on the bus for convenience of other possible clients (but I don't believe it is in use now).

krnowak replied:
Hm, I remember seeing in a review that introduced those changes that exporting this property was just a side-effect of adding a property which was meant to be used only internally (read: there was no `EosUpdaterData` struct then, which has this very purpose).

But sure, if you think that exporting this property might be useful for client then I'll just remove this commit.

cosimoc replied:
Yeah that was my review I think :-)
@dbnicholson convinced me that this property could be useful for other clients, so I'd rather leave it in.

dbnicholson replied:
I'll be honest that I originally did it because I wasn't sure the best way to store private data and that was the fastest way to make the property available. However, I do think it's useful for the client to see that this has changed, and it's also useful for debugging.

krnowak commented:
Updated, addressed some of the comments.

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -313,73 +323,193 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
+  *p_ref = g_key_file_get_string (bkf, group_name, OSTREE_REF_KEY, error);
+  if (*p_ref == NULL)
+    return FALSE;
+  return TRUE;
```
I think you should set `*on_hold = FALSE` before returning `TRUE` here. When the error is not set, all the out parameters should always be filled.

krnowak replied:
Sure.

cosimoc commented on src/eos-updater-util.c:
```diff
@@ -160,148 +153,165 @@ static const gchar *dmi_attributes[] =
-  if (!ostree_parse_refspec (o_refspec, &o_remote, &o_ref, error))
-    goto out;
+#define EOS_VENDOR_KEY "sys_vendor"
+#define EOS_PRODUCT_KEY "product_name"
```
I would define this without and `EOS` prefix... maybe `DMI_VENDOR_KEY` etc?

krnowak replied:
I'll rename it to `VENDOR_KEY`. The value also originates from device-tree stuff, so it is not really DMI-specific.

cosimoc commented on src/eos-updater-data.h:
```diff
@@ -0,0 +1,43 @@
+  OstreeRepo *repo;
+};
+
+void
```
These should also be in the same line as the declaration

cosimoc commented:
@krnowak Done another round of review, thanks.

krnowak commented:
Looks like github is acting up. Whenever I update my avahi branch which is based on this cleanups branch, it marks most of the comments as obsolete. Meh.

krnowak commented:
Updated. I removed the commit that moved the `original_refspec` variable to `EosUpdaterData` and added a small commit quieting down some make rules (so they print `GEN foo` when running `make V=0`).
